app/src/main/java/at/aau/edu/wizards/MainActivity.kt

```diff
@@ -142,16 +194,46 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
-    private fun handlePermissions() {
-        val permissionHandler = PermissionHandler(
-            activity = this,
-            permissions = REQUIRED_PERMISSIONS
-        ) { permissions ->
-            val anyDenied = permissions.any { !it.value }
-
-            println("Any permissions denied = $anyDenied")
-            // TODO: Need to properly handle denied permissions. App won't work without
+    private fun recheckPermissions(permission: String, name: String, requestCode: Int) {
+        when (PackageManager.PERMISSION_GRANTED) {
+            ContextCompat.checkSelfPermission(
+                applicationContext,
+                permission
+            ) -> {
+                checkPermission = true
+            }
+            else -> {
+                ActivityCompat.requestPermissions(
+                    this,
+                    arrayOf(permission),
+                    requestCode
+                )
+            }
+        }
+    }
+    override fun onRequestPermissionsResult(
```

Comment:
I mean it's not nice to have logic duplication now. But if you feel that works nice go for it